{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from vnpy.trader.vtConstant import *\n",
    "from vnpy.trader.app.ctaStrategy import CtaTemplate\n",
    "import numpy as np\n",
    "import talib as ta\n",
    "from datetime import timedelta, datetime\n",
    "import time\n",
    "\n",
    "########################################################################\n",
    "class adxMacdEosStrategy(CtaTemplate):\n",
    "    className = 'adxMacdEosStrategy'\n",
    "    author = 'ChannelCMT'\n",
    "\n",
    "    # 策略参数\n",
    "    barPeriod = 200\n",
    "    adxPeriod = 15; adxMaType = 0; adxThreshold = 20\n",
    "    fastPeriod = 30; slowPeriod = 65; signalPeriod = 30\n",
    "    fastMaType = 0; slowMaType = 0; signalMaType = 3\n",
    "    VWAPPeriod = 50\n",
    "    RSIPeriod=20\n",
    "    long=8\n",
    "\n",
    "    # 风控参数\n",
    "    trailingPct = 0.026; stopControlTime = 5\n",
    "    takeProfitFirstPct = 0.04; takeProfitSecondPct = 0.065\n",
    "    lot = 1800\n",
    "\n",
    "    # 仓位管理参数\n",
    "    addPct = 0.004; addMultipler = 1\n",
    "\n",
    "    # 信号变量\n",
    "    adxCanTrade = 0; priceDirection = 0\n",
    "    stopLossControl = 0\n",
    "\n",
    "    transactionPrice={}\n",
    "    # 自维护仓位与订单\n",
    "    ownPosDict = {}; orderDict = {}\n",
    "\n",
    "    # 参数列表，保存了参数的名称\n",
    "    paramList = [\n",
    "                 'symbolList',\n",
    "                 'adxPeriod', 'adxMaType', 'adxThreshold',\n",
    "                 'fastPeriod', 'slowPeriod','signalPeriod',\n",
    "                 'fastMaType', 'slowMaType', 'signalMaType',\n",
    "                 'trailingPct', 'stopControlTime',\n",
    "                 'addPct','addMultipler','lot',\n",
    "                 'takeProfitFirstPct', 'takeProfitSecondPct',\"long\"\n",
    "                 ]\n",
    "\n",
    "    # 变量列表，保存了变量的名称\n",
    "    varList = ['ownPosDict',\n",
    "               'adxCanTrade', 'priceDirection',\n",
    "               'transactionPrice',\n",
    "               'stopLossControl'\n",
    "               ]\n",
    "    # 同步列表，保存了需要保存到数据库的变量名称\n",
    "    syncList = ['posDict', 'eveningDict']\n",
    "\n",
    "    # ----------------------------------------------------------------------\n",
    "    def __init__(self, ctaEngine, setting):\n",
    "        super().__init__(ctaEngine, setting)\n",
    "\n",
    "    # ----------------------------------------------------------------------\n",
    "    def onInit(self):\n",
    "        \"\"\"初始化策略\"\"\"\n",
    "        self.writeCtaLog(u'策略初始化')\n",
    "        # signalVar\n",
    "        self.setArrayManagerSize(self.barPeriod)\n",
    "        self.tickObject = None\n",
    "\n",
    "        # riskControlVar\n",
    "        self.transactionPrice = {s+'0': 0 for s in self.symbolList}\n",
    "        self.closeTime = None\n",
    "        self.longStop = 0\n",
    "        self.shortStop = 999999\n",
    "        self.intraTradeHighDict = 0\n",
    "        self.intraTradeLowDict = 999999\n",
    "\n",
    "        # posManage\n",
    "        self.nPos = 0\n",
    "\n",
    "        for s in self.symbolList:\n",
    "            self.ownPosDict = {s + '_LONG': 0, s + '_SHORT': 0}\n",
    "            self.orderDict = {s + '_OPEN': [], s + '_CLOSE': []}\n",
    "\n",
    "        # 订单管理\n",
    "        self.toExcuteOrders = {}\n",
    "        self.toExcuteOrdersID = 0\n",
    "        self.mail(\"chushihuaaaaaaaaaaaaaaaaaaaaaaaaa\")\n",
    "        self.putEvent()\n",
    "\n",
    "    # ----------------------------------------------------------------------\n",
    "    def onStart(self):\n",
    "        \"\"\"启动策略（必须由用户继承实现）\"\"\"\n",
    "        self.writeCtaLog(u'策略启动')\n",
    "        self.putEvent()\n",
    "\n",
    "    # ----------------------------------------------------------------------\n",
    "    def onStop(self):\n",
    "        \"\"\"停止策略（必须由用户继承实现）\"\"\"\n",
    "        self.writeCtaLog(u'策略停止')\n",
    "        self.putEvent()\n",
    "\n",
    "    # ----------------------------------------------------------------------\n",
    "    def onTick(self, tick):\n",
    "        \"\"\"收到行情TICK推送\"\"\"\n",
    "        if not self.dataBlock(dataTime=tick.datetime, now=datetime.now(), maxDelay=5):\n",
    "            engineType = self.getEngineType()\n",
    "            if engineType == 'trading':\n",
    "                self.tickObject = tick\n",
    "            else:\n",
    "                pass\n",
    "\n",
    "    # 过滤掉实盘推数据可能产生的阻塞(5s延迟)\n",
    "    def dataBlock(self, dataTime, now, maxDelay=5):\n",
    "        if abs(now - dataTime).total_seconds() > maxDelay:\n",
    "            self.writeCtaLog(\n",
    "                \"数据推送阻塞,跳过该次推送:now=%s,dataTime=%s\" % (now.strftime(\"%Y-%m-%d %H:%M:%S\"),\n",
    "                                                      dataTime.strftime(\"%Y-%m-%d %H:%M:%S\")))\n",
    "            return True\n",
    "        else:\n",
    "            return False\n",
    "\n",
    "    # orderManagement--------------------------------------------------------\n",
    "    def cancelCloseOrder(self, bar):\n",
    "        symbol = bar.vtSymbol\n",
    "        haveCloseOrder = len(self.orderDict[symbol + '_CLOSE'])\n",
    "        if haveCloseOrder:\n",
    "            canceling = list(self.orderDict[symbol + '_CLOSE'])\n",
    "            for closeOrderId in canceling:\n",
    "                self.cancelOrder(closeOrderId)\n",
    "            return False, canceling\n",
    "        else:\n",
    "            return True, []\n",
    "\n",
    "    def priceExecute(self, bar):\n",
    "        engineType = self.getEngineType()\n",
    "        if engineType == 'trading':\n",
    "            buyExecute = self.tickObject.upperLimit*0.99\n",
    "            shortExecute = self.tickObject.lowerLimit*1.01\n",
    "        else:\n",
    "            buyExecute = bar.close * 1.02\n",
    "            shortExecute = bar.close * 0.98\n",
    "        return buyExecute, shortExecute\n",
    "\n",
    "    def buyCheckExtend(self, bar, volume=None):\n",
    "        symbol = bar.vtSymbol\n",
    "        if not volume:\n",
    "            volume = self.lot\n",
    "        buyExecute, shortExecute = self.priceExecute(bar)\n",
    "        if self.orderDict[symbol + '_OPEN']:\n",
    "            self.writeCtaLog('haveOpenOrder_Pass')\n",
    "        else:\n",
    "            self.buyOpen(symbol, buyExecute, volume)\n",
    "\n",
    "    def shortCheckExtend(self, bar, volume=None):\n",
    "        symbol = bar.vtSymbol\n",
    "        if not volume:\n",
    "            volume = self.lot\n",
    "        buyExecute, shortExecute = self.priceExecute(bar)\n",
    "        if self.orderDict[symbol + '_OPEN']:\n",
    "            self.writeCtaLog('haveOpenOrder_Pass')\n",
    "        else:\n",
    "            self.shortOpen(symbol, shortExecute, volume)\n",
    "\n",
    "    def coverCheckExtend(self, bar):\n",
    "        symbol = bar.vtSymbol\n",
    "        buyExecute, shortExecute = self.priceExecute(bar)\n",
    "        cancelled,canceling = self.cancelCloseOrder(bar)\n",
    "        if cancelled:\n",
    "            self.coverClose(symbol, buyExecute, self.ownPosDict[symbol + \"_SHORT\"])\n",
    "        else:\n",
    "            self.toExcuteOrdersID += 1\n",
    "            self.toExcuteOrders[self.toExcuteOrdersID] = {\n",
    "                \"symbol\": symbol,\n",
    "                \"price\": buyExecute,\n",
    "                \"volume\": self.ownPosDict[symbol + \"_SHORT\"],\n",
    "                \"orderType\": \"coverClose\",\n",
    "                \"canceling\":canceling\n",
    "            }\n",
    "\n",
    "    def sellCheckExtend(self, bar):\n",
    "        symbol = bar.vtSymbol\n",
    "        buyExecute, shortExecute = self.priceExecute(bar)\n",
    "        cancelled,canceling = self.cancelCloseOrder(bar)\n",
    "        if cancelled:\n",
    "            self.sellClose(symbol, shortExecute, self.ownPosDict[symbol + '_LONG'])\n",
    "        else:\n",
    "            self.toExcuteOrdersID += 1\n",
    "            self.toExcuteOrders[self.toExcuteOrdersID] = {\n",
    "                \"symbol\": symbol,\n",
    "                \"price\": shortExecute,\n",
    "                \"volume\": self.ownPosDict[symbol + '_LONG'],\n",
    "                \"orderType\": \"sellClose\",\n",
    "                \"canceling\": canceling\n",
    "            }\n",
    "\n",
    "    def shortOpen(self, symbol, price, volume):\n",
    "        shortOpenOrderList = self.short(symbol, price, volume)\n",
    "        self.orderDict[symbol + '_OPEN'].extend(shortOpenOrderList)\n",
    "\n",
    "    def buyOpen(self, symbol, price, volume):\n",
    "        buyOpenOrderList = self.buy(symbol, price, volume)\n",
    "        self.orderDict[symbol + '_OPEN'].extend(buyOpenOrderList)\n",
    "\n",
    "    def sellClose(self, symbol, price, volume):\n",
    "        sellCloseOrderList = self.sell(symbol, price, volume)\n",
    "        self.orderDict[symbol + '_CLOSE'].extend(sellCloseOrderList)\n",
    "\n",
    "    def coverClose(self, symbol, price, volume):\n",
    "        coverCloseOrderList = self.cover(symbol, price, volume)\n",
    "        self.orderDict[symbol + '_CLOSE'].extend(coverCloseOrderList)\n",
    "\n",
    "\n",
    "# executeManagement--------------------------------------------------------\n",
    "#     def on5sBar(self, bar):\n",
    "    def onBar(self, bar):\n",
    "        self.writeCtaLog('adxMacdEosStrategy####5S####ownposDict:%s####'%(self.ownPosDict))\n",
    "        self.onBarRiskControl(bar)\n",
    "        self.onBarExitTimeControl(bar)\n",
    "        self.onBarExecute(bar)\n",
    "        self.onBarPosition(bar)\n",
    "\n",
    "    def onBarExit(self, bar):\n",
    "        symbol = bar.vtSymbol\n",
    "        if self.priceDirection == -1:\n",
    "            if self.ownPosDict[symbol + \"_LONG\"] > 0 :\n",
    "                self.sellCheckExtend(bar)\n",
    "        elif self.priceDirection == 1:\n",
    "            if self.ownPosDict[symbol + \"_SHORT\"] > 0:\n",
    "                self.coverCheckExtend(bar)\n",
    "\n",
    "    def onBarExitTimeControl(self, bar):\n",
    "        if self.closeTime:\n",
    "            if (bar.datetime - self.closeTime) >= timedelta(hours=self.stopControlTime):\n",
    "                self.stopLossControl = 0\n",
    "\n",
    "    def onBarRiskControl(self, bar):\n",
    "        symbol = bar.vtSymbol\n",
    "        if self.ownPosDict[symbol + \"_LONG\"] == 0 and self.ownPosDict[symbol + \"_SHORT\"] == 0:\n",
    "            self.intraTradeHighDict = 0\n",
    "            self.intraTradeLowDict = 999999\n",
    "            self.longStop = 0\n",
    "            self.shortStop = 999999\n",
    "            self.nChange = 0\n",
    "\n",
    "        # 持有多头仓位\n",
    "        elif self.ownPosDict[symbol + \"_LONG\"] > 0:\n",
    "            firstOrder=self.transactionPrice[symbol+'0']\n",
    "            self.intraTradeHighDict = max(self.intraTradeHighDict, bar.high)\n",
    "            self.nChange = (self.intraTradeHighDict/firstOrder-1)//self.trailingPct\n",
    "            changePrice = firstOrder*self.nChange*self.trailingPct\n",
    "            self.longStop = max(self.longStop, firstOrder*(1-self.trailingPct)+changePrice)\n",
    "            if bar.low <= self.longStop:\n",
    "                self.sellCheckExtend(bar)\n",
    "            self.writeCtaLog('longStop%s'%(self.longStop))\n",
    "        # 持有空头仓位\n",
    "        elif self.ownPosDict[symbol + \"_SHORT\"] > 0:\n",
    "            firstOrder=self.transactionPrice[symbol+'0']\n",
    "            self.intraTradeLowDict = min(self.intraTradeLowDict, bar.low)\n",
    "            self.nChange = -1* (self.intraTradeLowDict/firstOrder-1)//self.trailingPct\n",
    "            changePrice = firstOrder*self.nChange*self.trailingPct\n",
    "            self.shortStop = min(self.shortStop, firstOrder*(1+self.trailingPct)-changePrice)\n",
    "            if bar.high >= self.shortStop:\n",
    "                self.coverCheckExtend(bar)\n",
    "            self.writeCtaLog('shortStop%s'%(self.shortStop))\n",
    "\n",
    "    def onBarExecute(self, bar):\n",
    "        symbol = bar.vtSymbol\n",
    "        if (self.adxCanTrade == 1):\n",
    "            if (self.priceDirection == 1) and (self.ownPosDict[symbol + \"_LONG\"] == 0):\n",
    "                if self.stopLossControl == -1:\n",
    "                    self.stopLossControl = 0\n",
    "                if (self.stopLossControl == 0):\n",
    "                    if (self.ownPosDict[symbol + \"_SHORT\"] == 0):\n",
    "                        self.buyCheckExtend(bar)\n",
    "                    elif (self.ownPosDict[symbol + \"_SHORT\"] > 0):\n",
    "                        self.coverCheckExtend(bar)\n",
    "                        self.buyCheckExtend(bar)\n",
    "\n",
    "            elif (self.priceDirection == -1) and (self.ownPosDict[symbol + \"_SHORT\"] == 0):\n",
    "                if self.stopLossControl == 1:\n",
    "                    self.stopLossControl = 0\n",
    "                if (self.stopLossControl == 0):\n",
    "                    if (self.ownPosDict[symbol + \"_LONG\"] == 0):\n",
    "                        self.shortCheckExtend(bar)\n",
    "                    elif (self.ownPosDict[symbol + \"_LONG\"] > 0):\n",
    "                        self.sellCheckExtend(bar)\n",
    "                        self.shortCheckExtend(bar)\n",
    "            else:\n",
    "                self.onBarExit(bar)\n",
    "        else:\n",
    "            self.onBarExit(bar)\n",
    "        self.putEvent()\n",
    "\n",
    "    def onBarPosition(self,bar):\n",
    "        symbol = bar.vtSymbol\n",
    "        firstOrder=self.transactionPrice[symbol+'0']\n",
    "        if (self.ownPosDict[symbol+'_LONG']==0) and (self.ownPosDict[symbol + \"_SHORT\"]==0):\n",
    "            self.nPos=0\n",
    "        elif (self.ownPosDict[symbol+'_LONG']>0 and self.nPos < 1):\n",
    "            if (firstOrder/bar.close-1) >= self.addPct:\n",
    "                self.nPos += 1\n",
    "                self.buyCheckExtend(bar, int(self.lot*(self.addMultipler**self.nPos)))\n",
    "        elif (self.ownPosDict[symbol + \"_SHORT\"] > 0 and self.nPos < 1):\n",
    "            if (bar.close/firstOrder-1) >= self.addPct:\n",
    "                self.nPos += 1\n",
    "                self.shortCheckExtend(bar, int(self.lot*(self.addMultipler**self.nPos)))\n",
    "\n",
    "\n",
    "    # ----------------------------------------------------------------------\n",
    " \n",
    "    def on30MinBar(self, bar):\n",
    "        symbol = bar.vtSymbol\n",
    "\n",
    "        am30 = self.getArrayManager(symbol, \"30m\")\n",
    "\n",
    "        if not am30.inited:\n",
    "            return\n",
    "\n",
    "        adxTrend = ta.ADX(am30.high, am30.low, am30.close, self.adxPeriod)\n",
    "        VWAP = ta.SUM(am30.close*am30.volume, self.VWAPPeriod) / ta.SUM(am30.volume, self.VWAPPeriod)\n",
    "\n",
    "# 现象条件\n",
    "        breakUpVwap = (am30.close[-1] > VWAP[-2]) and (am30.close[-2] <= VWAP[-2])\n",
    "        breakDnVwap = (am30.close[-1] < VWAP[-2]) and (am30.close[-2] >= VWAP[-2])\n",
    "        more_volume=((am30.volume[-2]+am30.volume[-3]+am30.volume[-4])/3)>((am30.volume[-4]+am30.volume[-5]+am30.volume[-6])/3)\n",
    "        HH=max(am30.close[-10:-2])\n",
    "        LL=min(am30.low[-10:-2])\n",
    "        mom=am30.close[-1]>am30.close[-self.long]\n",
    "        if(more_volume) and (mom>0 or mom<0):\n",
    "            self.adxCanTrade = 1\n",
    "        else:\n",
    "            self.adxCanTrade = -1\n",
    "\n",
    "        self.writeCtaLog('adxTrend: %s, adxCanTrade: %s:'%(adxTrend[-1], self.adxCanTrade))\n",
    "\n",
    "        macd, macdSignal, macdHist = ta.MACDEXT(am30.close, self.fastPeriod, self.fastMaType, \\\n",
    "                                                self.slowPeriod, self.slowMaType, self.signalPeriod, self.signalMaType)\n",
    "\n",
    "        if (macd[-1]>macdSignal[-1]) and (macd[-1]>macd[-3]):\n",
    "            self.priceDirection = 1\n",
    "        elif (macd[-1]<macdSignal[-1]) and (macd[-1]<macd[-3]):\n",
    "            self.priceDirection = -1\n",
    "        else:\n",
    "            self.priceDirection = 0\n",
    "\n",
    "\n",
    "        #self.writeCtaLog(u'macd: %s, macdSignal: %s, priceDirection: %s'%(macd[-3:], macdSignal[-1], self.priceDirection))\n",
    "        self.putEvent()\n",
    "\n",
    "    # ----------------------------------------------------------------------\n",
    "    def dealtoExcuteOrders(self, symbol):\n",
    "        for ID in list(self.toExcuteOrders):\n",
    "            order = self.toExcuteOrders[ID]\n",
    "            if order[\"orderType\"] in [\"coverClose\", \"sellClose\"]:\n",
    "                # 前置要求中待撤销的订单已从orderDict中全部撤销\n",
    "                if len(set(order[\"canceling\"]) & set(self.orderDict[symbol + '_CLOSE'])) == 0:\n",
    "                    if order[\"orderType\"] == \"coverClose\":\n",
    "                        self.coverClose(order[\"symbol\"], order[\"price\"], order[\"volume\"])\n",
    "                    elif order[\"orderType\"] == \"sellClose\":\n",
    "                        self.sellClose(order[\"symbol\"], order[\"price\"], order[\"volume\"])\n",
    "                    # 清除该待执行订单\n",
    "                    del self.toExcuteOrders[ID]\n",
    "\n",
    "    # ----------------------------------------------------------------------\n",
    "    def onOrder(self, order):\n",
    "        symbol = order.vtSymbol\n",
    "        self.onOrderEmail(order)\n",
    "\n",
    "        if order.status in STATUS_FINISHED:\n",
    "            if (order.offset == OFFSET_OPEN) and (str(order.vtOrderID) in self.orderDict[symbol + '_OPEN']):\n",
    "                self.orderDict[symbol + '_OPEN'].remove(str(order.vtOrderID))\n",
    "            elif (order.offset == OFFSET_CLOSE) and (str(order.vtOrderID) in self.orderDict[symbol + '_CLOSE']):\n",
    "                self.orderDict[symbol + '_CLOSE'].remove(str(order.vtOrderID))\n",
    "        # 触发撤单成功,扫描待执行订单的前置撤单要求是否达到，达到则触发对待执行订单的发单\n",
    "        if order.status == STATUS_CANCELLED:\n",
    "            self.dealtoExcuteOrders(symbol)\n",
    "\n",
    "    def onOrderEmail(self, order):\n",
    "        if order.status == STATUS_UNKNOWN:\n",
    "            self.mail(u'出现未知订单，需要策略师外部干预,ID:%s, symbol:%s,direction:%s,offset:%s'\n",
    "                      % (order.vtOrderID, order.vtSymbol, order.direction, order.offset))\n",
    "        if order.status == STATUS_REJECTED:\n",
    "            self.mail(u'Rejected,ID:%s, symbol:%s,direction:%s,offset:%s,拒单信息:%s'\n",
    "                      % (order.vtOrderID, order.vtSymbol, order.direction, order.offset,order.rejectedInfo))\n",
    "        if order.thisTradedVolume != 0:\n",
    "            content = u'成交信息播报,ID:%s, symbol:%s, directionL%s, offset:%s, price:%s'%\\\n",
    "                      (order.vtOrderID, order.vtSymbol, order.direction, order.offset, order.price)\n",
    "            self.mail(content)\n",
    "\n",
    "    # ----------------------------------------------------------------------\n",
    "    def onTrade(self, trade):\n",
    "        symbol = trade.vtSymbol\n",
    "        self.writeCtaLog('tradeTime:%s,offset:%s,transactionPrice:%s ,ownPosDict%s'\\\n",
    "                        %(trade.tradeDatetime, trade.offset, self.transactionPrice, self.ownPosDict))\n",
    "        if trade.offset == OFFSET_OPEN:\n",
    "            self.transactionPrice[symbol+'%s'%(self.nPos)] = trade.price\n",
    "        elif trade.offset == OFFSET_CLOSE:\n",
    "            self.closeTime = trade.tradeDatetime\n",
    "            if trade.direction == DIRECTION_SHORT:\n",
    "                self.stopLossControl = 1\n",
    "            elif trade.direction == DIRECTION_LONG:\n",
    "                self.stopLossControl = -1\n",
    "        self.onTradeOwnPosDict(trade)\n",
    "        self.takeProfit(trade)\n",
    "\n",
    "    def onTradeOwnPosDict(self, trade):\n",
    "        symbol = trade.vtSymbol\n",
    "        if trade.direction == DIRECTION_LONG and trade.offset == OFFSET_OPEN:\n",
    "            self.ownPosDict[symbol + '_LONG'] += int(trade.volume)\n",
    "        elif trade.direction == DIRECTION_SHORT and trade.offset == OFFSET_CLOSE:\n",
    "            self.ownPosDict[symbol + '_LONG'] -= int(trade.volume)\n",
    "        elif trade.direction == DIRECTION_SHORT and trade.offset == OFFSET_OPEN:\n",
    "            self.ownPosDict[symbol + '_SHORT'] += int(trade.volume)\n",
    "        elif trade.direction == DIRECTION_LONG and trade.offset == OFFSET_CLOSE:\n",
    "            self.ownPosDict[symbol + '_SHORT'] -= int(trade.volume)\n",
    "\n",
    "    def takeProfit(self, trade):\n",
    "        symbol = trade.vtSymbol\n",
    "        firstOrder = self.transactionPrice[symbol+'0']\n",
    "        longFirstProfit = firstOrder*(1+self.takeProfitFirstPct)\n",
    "        longSecondProfit = firstOrder*(1+self.takeProfitSecondPct)\n",
    "        shortFirstProfit = firstOrder*(1-self.takeProfitFirstPct)\n",
    "        shortSecondProfit = firstOrder*(1-self.takeProfitSecondPct)\n",
    "\n",
    "        if trade.offset == OFFSET_OPEN:\n",
    "            if trade.direction == DIRECTION_LONG:\n",
    "                self.sellClose(symbol, longFirstProfit, trade.volume//2)\n",
    "                self.sellClose(symbol, longSecondProfit, trade.volume-trade.volume//2)\n",
    "                self.writeCtaLog('long### tp1:%s, tp2:%s'%(longFirstProfit, longSecondProfit))\n",
    "            elif trade.direction == DIRECTION_SHORT:\n",
    "                self.coverClose(symbol, shortFirstProfit, trade.volume//2)\n",
    "                self.coverClose(symbol, shortSecondProfit,trade.volume-trade.volume//2)\n",
    "                self.writeCtaLog('short### tp1:%s, tp2:%s'%(shortFirstProfit, shortSecondProfit))\n",
    "    # ----------------------------------------------------------------------\n",
    "    def onStopOrder(self, so):\n",
    "        \"\"\"停止单推送\"\"\"\n",
    "        pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "仓位字典构造完成 \n",
      "初始仓位: {'eos_quarter:OKEX_LONG': 0, 'eos_quarter:OKEX_SHORT': 0}\n",
      "2019-01-19 18:07:24.735713\t开始回测\n",
      "2019-01-19 18:07:24.735914\t策略初始化\n",
      "2019-01-19 18:07:25.188165\t数据载入完成, 时间段:[20180927 08:00,20181001 12:00);数据量:5266\n",
      "2019-01-19 18:07:25.188447\t初始化预加载数据成功, 数据长度:5266\n",
      "2019-01-19 18:07:25.188557\t策略初始化完成\n",
      "2019-01-19 18:07:25.188598\t策略启动完成\n",
      "2019-01-19 18:07:25.188666\t回测时间范围:[20181001 12:00,20190110 16:00)\n",
      "2019-01-19 18:07:25.188743\t当前回放的时间段:[20181001 12:00,20190110 16:00)\n",
      "2019-01-19 18:07:35.844131\t数据载入完成, 时间段:[20181001 12:00,20190110 16:00);数据量:144434\n",
      "2019-01-19 18:07:36.189149\t数据载入完成, 时间段:[20180930 05:51,20181003 12:01);数据量:4689\n",
      "2019-01-19 18:07:39.322969\t数据载入完成, 时间段:[20180824 19:30,20180930 05:50);数据量:38886\n",
      "2019-01-19 18:07:39.329037\tWARNING: 该时间段:[20181003 12:01,20181003 12:30) 数据量为0!\n",
      "2019-01-19 18:08:04.963831\t回放结束ress: 100%    \n"
     ]
    }
   ],
   "source": [
    "from vnpy.trader.app.ctaStrategy import BacktestingEngine\n",
    "import pandas as pd\n",
    "from vnpy.trader.utils import htmlplot\n",
    "\n",
    "def runBacktesting(strategyClass, settingDict,\n",
    "                   startDate, endDate, slippage, rate):\n",
    "    engine = BacktestingEngine()\n",
    "    engine.setBacktestingMode(engine.BAR_MODE)\n",
    "    engine.setDatabase('VnTrader_1Min_Db')\n",
    "    engine.setStartDate(startDate, initHours=100)\n",
    "    engine.setEndDate(endDate)\n",
    "    engine.setSlippage(slippage)\n",
    "    engine.setRate(rate)\n",
    "    engine.setLog(True)\n",
    "    \n",
    "#   在这里设置画图参数，目前只用设置freqency，支持的有 s(second) m(minute) h(hour) d(day)，也可以用timestamp, 例如： 1h -> timestamp(hours=1)\n",
    "#   sample(输出用4hbar画图)\n",
    "#   画图需要安装bokeh==0.12.14\n",
    "    engine.initStrategy(strategyClass, settingDict)\n",
    "    engine.setCapital(200000)\n",
    "    engine.runBacktesting()\n",
    "    #显示逐日回测结果\n",
    "    engine.showDailyResult()\n",
    "    #显示逐笔回测结果\n",
    "    engine.showBacktestingResult()\n",
    "    htmlplot.showTransaction(engine, frequency=\"1h\")\n",
    "    # 计算回测结果\n",
    "    perfromance = engine.calculateDailyResult()\n",
    "    perfromanceDf , result = engine.calculateDailyStatistics(perfromance)\n",
    "    tradeReport = pd.DataFrame([obj.__dict__ for obj in engine.tradeDict.values()])\n",
    "    tradeDf = tradeReport.set_index('tradeDatetime')\n",
    "    return engine\n",
    "#     return perfromanceDf, tradeDf\n",
    "\n",
    "runBacktesting(adxMacdEosStrategy, {'symbolList': [\n",
    "                                                       # 'EOSUSDT:binance',\n",
    "                                                             'eos_quarter:OKEX'\n",
    "                                                        ]},\n",
    "                                                           '20181001 12:00', \n",
    "#                                                                '20181006 12:00', \n",
    "                                                           '20190110 16:00',0.002, 5/10000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "80"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import numpy as np\n",
    "ara=[80,11,12,22,20,41,32,55,8,9]\n",
    "HH=max(ara[-11:-1])\n",
    "HH\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 15:01:14.748929\t开始回测\n",
      "2019-01-19 15:01:14.749934\t策略初始化\n",
      "2019-01-19 15:01:15.321398\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 15:01:15.321398\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 15:01:15.321398\t策略初始化完成\n",
      "2019-01-19 15:01:15.321398\t策略启动完成\n",
      "2019-01-19 15:01:15.322396\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:01:15.322396\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:01:56.332076\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:01:56.407780\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 15:01:56.408776\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:01:56.649106\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 15:02:29.270624\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:02:29.284489\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 15:02:29.292438\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:03:05.480711\t回放结束ress: 100%    \n"
     ]
    }
   ],
   "source": [
    "if __name__==\"__main__\":\n",
    "    from vnpy.trader.app.ctaStrategy import BacktestingEngine\n",
    "\n",
    "    # 创建回测引擎对象\n",
    "    engine = BacktestingEngine()\n",
    "\n",
    "    # 设置回测使用的数据\n",
    "    engine.setBacktestingMode(engine.BAR_MODE)    # 设置引擎的回测模式为K线\n",
    "    engine.setDatabase('VnTrader_1Min_Db')  # 设置使用的历史数据库\n",
    "    engine.setStartDate('20180701 12:00',initHours=200) # 设置回测用的数据起始日期\n",
    "    engine.setEndDate('20181201 12:00') # 设置回测用的数据终止日期\n",
    "\n",
    "    # 配置回测引擎参数\n",
    "    engine.setSlippage(0.002)    # 设置滑点\n",
    "    engine.setRate(5/10000)   # 设置手续费千1\n",
    "    engine.setCapital(200000)  # 设置回测本金\n",
    "\n",
    "    # # 在引擎中创建策略对象\n",
    "    parameterDict = {'symbolList':['EOSUSDT:binance']}          # 策略参数配置\n",
    "    engine.initStrategy(adxMacdEosStrategy, parameterDict)    # 创建策略对象\n",
    "    engine.runBacktesting()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:27:52.997067\t------------------------------\n",
      "2019-01-19 15:27:52.998068\tsetting: {'fastPeriod': 20, 'slowPeriod': 40, 'signalMaPeriod': 10, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 15:27:53.050924\t开始回测\n",
      "2019-01-19 15:27:53.050924\t策略初始化\n",
      "2019-01-19 15:27:53.618407\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 15:27:53.618407\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 15:27:53.618407\t策略初始化完成\n",
      "2019-01-19 15:27:53.618407\t策略启动完成\n",
      "2019-01-19 15:27:53.618407\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:27:53.618407\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:28:33.776910\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:28:33.855578\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 15:28:33.855578\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:28:34.116880\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 15:29:06.566372\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:29:06.582201\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 15:29:06.591177\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:29:41.210649\t回放结束ress: 100%    \n",
      "2019-01-19 15:29:41.210649\t计算按日统计结果\n",
      "2019-01-19 15:29:41.342290\t------------------------------\n",
      "2019-01-19 15:29:41.342290\tsetting: {'fastPeriod': 20, 'slowPeriod': 40, 'signalMaPeriod': 15, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 15:29:41.365230\t开始回测\n",
      "2019-01-19 15:29:41.365230\t策略初始化\n",
      "2019-01-19 15:29:41.909774\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 15:29:41.909774\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 15:29:41.909774\t策略初始化完成\n",
      "2019-01-19 15:29:41.909774\t策略启动完成\n",
      "2019-01-19 15:29:41.909774\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:29:41.909774\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:30:21.882857\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:30:21.989569\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 15:30:21.990565\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:30:22.286772\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 15:30:54.809794\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:30:54.836719\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 15:30:54.844666\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:31:29.687536\t回放结束ress: 100%    \n",
      "2019-01-19 15:31:29.687536\t计算按日统计结果\n",
      "2019-01-19 15:31:29.876040\t------------------------------\n",
      "2019-01-19 15:31:29.876040\tsetting: {'fastPeriod': 20, 'slowPeriod': 40, 'signalMaPeriod': 20, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 15:31:29.922908\t开始回测\n",
      "2019-01-19 15:31:29.924915\t策略初始化\n",
      "2019-01-19 15:31:30.633009\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 15:31:30.634006\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 15:31:30.634006\t策略初始化完成\n",
      "2019-01-19 15:31:30.634006\t策略启动完成\n",
      "2019-01-19 15:31:30.634006\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:31:30.634006\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:33:10.722294\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:33:10.823048\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 15:33:10.824041\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:33:11.090303\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 15:33:43.726140\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:33:43.754942\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 15:33:43.763916\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:34:18.631760\t回放结束ress: 100%    \n",
      "2019-01-19 15:34:18.633756\t计算按日统计结果\n",
      "2019-01-19 15:34:18.838207\t------------------------------\n",
      "2019-01-19 15:34:18.838207\tsetting: {'fastPeriod': 20, 'slowPeriod': 60, 'signalMaPeriod': 10, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 15:34:18.862156\t开始回测\n",
      "2019-01-19 15:34:18.862156\t策略初始化\n",
      "2019-01-19 15:34:19.411707\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 15:34:19.411707\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 15:34:19.411707\t策略初始化完成\n",
      "2019-01-19 15:34:19.411707\t策略启动完成\n",
      "2019-01-19 15:34:19.411707\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:34:19.411707\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:35:00.665344\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:35:00.725089\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 15:35:00.725089\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:35:00.976389\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 15:35:33.368607\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:35:33.384565\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 15:35:33.391563\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:35:43.183372\tProgress: 22%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py:649: UserWarning: 当前平仓数量大于可平量，实盘下可能拒单, 请小心处理,下单信息为---direction:买平;volume:1000;pos=0\n",
      "  warnings.warn(\"当前平仓数量大于可平量，实盘下可能拒单, 请小心处理,下单信息为---direction:买平;volume:%s;pos=%s\"%(order.totalVolume, closable))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:36:06.960818\t回放结束ress: 100%    \n",
      "2019-01-19 15:36:06.960818\t计算按日统计结果\n",
      "2019-01-19 15:36:07.108422\t------------------------------\n",
      "2019-01-19 15:36:07.109423\tsetting: {'fastPeriod': 20, 'slowPeriod': 60, 'signalMaPeriod': 15, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 15:36:07.137345\t开始回测\n",
      "2019-01-19 15:36:07.137345\t策略初始化\n",
      "2019-01-19 15:36:07.801570\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 15:36:07.801570\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 15:36:07.801570\t策略初始化完成\n",
      "2019-01-19 15:36:07.801570\t策略启动完成\n",
      "2019-01-19 15:36:07.801570\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:36:07.801570\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:36:49.518204\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:36:49.590883\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 15:36:49.590883\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:36:49.827253\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 15:37:22.302631\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:37:22.320586\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 15:37:22.333549\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:38:01.073998\t回放结束ress: 100%    \n",
      "2019-01-19 15:38:01.073998\t计算按日统计结果\n",
      "2019-01-19 15:38:01.209635\t------------------------------\n",
      "2019-01-19 15:38:01.210633\tsetting: {'fastPeriod': 20, 'slowPeriod': 60, 'signalMaPeriod': 20, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 15:38:01.235566\t开始回测\n",
      "2019-01-19 15:38:01.235566\t策略初始化\n",
      "2019-01-19 15:38:01.770153\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 15:38:01.770153\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 15:38:01.770153\t策略初始化完成\n",
      "2019-01-19 15:38:01.770153\t策略启动完成\n",
      "2019-01-19 15:38:01.771134\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:38:01.771134\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:39:39.720456\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:39:39.865029\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 15:39:39.866026\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:39:40.178218\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 15:40:13.349262\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:40:13.361229\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 15:40:13.371202\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:40:49.093719\t回放结束ress: 100%    \n",
      "2019-01-19 15:40:49.093719\t计算按日统计结果\n",
      "2019-01-19 15:40:49.290204\t------------------------------\n",
      "2019-01-19 15:40:49.290204\tsetting: {'fastPeriod': 20, 'slowPeriod': 80, 'signalMaPeriod': 10, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 15:40:49.315126\t开始回测\n",
      "2019-01-19 15:40:49.315126\t策略初始化\n",
      "2019-01-19 15:40:50.135934\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 15:40:50.136931\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 15:40:50.136931\t策略初始化完成\n",
      "2019-01-19 15:40:50.136931\t策略启动完成\n",
      "2019-01-19 15:40:50.136931\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:40:50.137930\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:41:33.500645\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:41:33.563422\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 15:41:33.563422\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:41:33.820735\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 15:42:06.521478\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:42:06.556269\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 15:42:06.569228\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:42:42.955971\t回放结束ress: 100%    \n",
      "2019-01-19 15:42:42.955971\t计算按日统计结果\n",
      "2019-01-19 15:42:43.111552\t------------------------------\n",
      "2019-01-19 15:42:43.111552\tsetting: {'fastPeriod': 20, 'slowPeriod': 80, 'signalMaPeriod': 15, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 15:42:43.137484\t开始回测\n",
      "2019-01-19 15:42:43.137484\t策略初始化\n",
      "2019-01-19 15:42:43.678039\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 15:42:43.678039\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 15:42:43.678039\t策略初始化完成\n",
      "2019-01-19 15:42:43.678039\t策略启动完成\n",
      "2019-01-19 15:42:43.678039\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:42:43.678039\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:43:24.901939\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:43:24.984619\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 15:43:24.984619\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:43:25.223959\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 15:43:57.670077\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:43:57.687907\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 15:43:57.695885\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:44:33.999849\t回放结束ress: 100%    \n",
      "2019-01-19 15:44:33.999849\t计算按日统计结果\n",
      "2019-01-19 15:44:34.224248\t------------------------------\n",
      "2019-01-19 15:44:34.224248\tsetting: {'fastPeriod': 20, 'slowPeriod': 80, 'signalMaPeriod': 20, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 15:44:34.249182\t开始回测\n",
      "2019-01-19 15:44:34.249182\t策略初始化\n",
      "2019-01-19 15:44:34.871518\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 15:44:34.871518\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 15:44:34.871518\t策略初始化完成\n",
      "2019-01-19 15:44:34.871518\t策略启动完成\n",
      "2019-01-19 15:44:34.871518\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:44:34.871518\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:45:15.525004\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:45:15.596721\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 15:45:15.596721\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:45:15.831064\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 15:45:48.291220\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:45:48.316114\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 15:45:48.324092\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:46:21.859454\t回放结束ress: 100%    \n",
      "2019-01-19 15:46:21.859454\t计算按日统计结果\n",
      "2019-01-19 15:46:21.986115\t------------------------------\n",
      "2019-01-19 15:46:21.986115\tsetting: {'fastPeriod': 30, 'slowPeriod': 40, 'signalMaPeriod': 10, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 15:46:22.011049\t开始回测\n",
      "2019-01-19 15:46:22.011049\t策略初始化\n",
      "2019-01-19 15:46:22.531658\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 15:46:22.531658\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 15:46:22.531658\t策略初始化完成\n",
      "2019-01-19 15:46:22.531658\t策略启动完成\n",
      "2019-01-19 15:46:22.531658\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:46:22.531658\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:48:07.590160\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:48:07.668815\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 15:48:07.668815\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:48:07.930117\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 15:48:40.660481\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:48:40.682327\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 15:48:40.690278\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:49:20.154794\t回放结束ress: 100%    \n",
      "2019-01-19 15:49:20.156789\t计算按日统计结果\n",
      "2019-01-19 15:49:20.440031\t------------------------------\n",
      "2019-01-19 15:49:20.440031\tsetting: {'fastPeriod': 30, 'slowPeriod': 40, 'signalMaPeriod': 15, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 15:49:20.469951\t开始回测\n",
      "2019-01-19 15:49:20.469951\t策略初始化\n",
      "2019-01-19 15:49:21.071344\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 15:49:21.071344\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 15:49:21.072341\t策略初始化完成\n",
      "2019-01-19 15:49:21.072341\t策略启动完成\n",
      "2019-01-19 15:49:21.072341\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:49:21.072341\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:50:02.010916\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:50:02.087710\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 15:50:02.087710\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:50:02.331060\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 15:50:34.835724\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:50:34.848674\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 15:50:34.855654\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:51:16.664903\t回放结束ress: 100%    \n",
      "2019-01-19 15:51:16.664903\t计算按日统计结果\n",
      "2019-01-19 15:51:16.827467\t------------------------------\n",
      "2019-01-19 15:51:16.827467\tsetting: {'fastPeriod': 30, 'slowPeriod': 40, 'signalMaPeriod': 20, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 15:51:16.855393\t开始回测\n",
      "2019-01-19 15:51:16.855393\t策略初始化\n",
      "2019-01-19 15:51:17.421879\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 15:51:17.421879\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 15:51:17.421879\t策略初始化完成\n",
      "2019-01-19 15:51:17.421879\t策略启动完成\n",
      "2019-01-19 15:51:17.422880\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:51:17.422880\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:51:58.877101\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:51:58.936914\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 15:51:58.937911\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:51:59.179266\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 15:52:31.564729\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:52:31.577668\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 15:52:31.585655\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:53:09.859346\t回放结束ress: 100%    \n",
      "2019-01-19 15:53:09.860342\t计算按日统计结果\n",
      "2019-01-19 15:53:10.009942\t------------------------------\n",
      "2019-01-19 15:53:10.010939\tsetting: {'fastPeriod': 30, 'slowPeriod': 60, 'signalMaPeriod': 10, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 15:53:10.037867\t开始回测\n",
      "2019-01-19 15:53:10.037867\t策略初始化\n",
      "2019-01-19 15:53:10.599367\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 15:53:10.599367\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 15:53:10.599367\t策略初始化完成\n",
      "2019-01-19 15:53:10.599367\t策略启动完成\n",
      "2019-01-19 15:53:10.599367\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:53:10.599367\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:55:04.152848\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:55:04.253581\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 15:55:04.254584\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:55:04.578711\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 15:55:37.487747\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:55:37.502707\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 15:55:37.523652\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:56:12.500164\t回放结束ress: 100%    \n",
      "2019-01-19 15:56:12.500164\t计算按日统计结果\n",
      "2019-01-19 15:56:12.689657\t------------------------------\n",
      "2019-01-19 15:56:12.689657\tsetting: {'fastPeriod': 30, 'slowPeriod': 60, 'signalMaPeriod': 15, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 15:56:12.715587\t开始回测\n",
      "2019-01-19 15:56:12.715587\t策略初始化\n",
      "2019-01-19 15:56:13.274094\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 15:56:13.274094\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 15:56:13.274094\t策略初始化完成\n",
      "2019-01-19 15:56:13.274094\t策略启动完成\n",
      "2019-01-19 15:56:13.275092\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:56:13.275092\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:56:54.550765\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:56:54.612601\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 15:56:54.613598\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:56:54.855950\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 15:57:27.318401\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:57:27.334305\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 15:57:27.341287\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:58:02.789536\t回放结束ress: 100%    \n",
      "2019-01-19 15:58:02.789536\t计算按日统计结果\n",
      "2019-01-19 15:58:02.943153\t------------------------------\n",
      "2019-01-19 15:58:02.944155\tsetting: {'fastPeriod': 30, 'slowPeriod': 60, 'signalMaPeriod': 20, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 15:58:02.971051\t开始回测\n",
      "2019-01-19 15:58:02.971051\t策略初始化\n",
      "2019-01-19 15:58:03.520582\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 15:58:03.520582\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 15:58:03.521580\t策略初始化完成\n",
      "2019-01-19 15:58:03.521580\t策略启动完成\n",
      "2019-01-19 15:58:03.521580\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:58:03.521580\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:58:44.174329\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:58:44.253070\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 15:58:44.253070\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:58:44.486445\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 15:59:17.058671\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 15:59:17.076500\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 15:59:17.083480\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 15:59:53.863173\t回放结束ress: 100%    \n",
      "2019-01-19 15:59:53.864172\t计算按日统计结果\n",
      "2019-01-19 15:59:54.010776\t------------------------------\n",
      "2019-01-19 15:59:54.011783\tsetting: {'fastPeriod': 30, 'slowPeriod': 80, 'signalMaPeriod': 10, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 15:59:54.040714\t开始回测\n",
      "2019-01-19 15:59:54.040714\t策略初始化\n",
      "2019-01-19 15:59:54.624137\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 15:59:54.624137\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 15:59:54.624137\t策略初始化完成\n",
      "2019-01-19 15:59:54.624137\t策略启动完成\n",
      "2019-01-19 15:59:54.624137\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 15:59:54.624137\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 16:00:35.462224\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 16:00:35.522948\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 16:00:35.523945\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 16:00:35.754356\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 16:01:08.189857\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 16:01:08.230747\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 16:01:08.260663\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 16:01:49.758744\t回放结束ress: 100%    \n",
      "2019-01-19 16:01:49.759741\t计算按日统计结果\n",
      "2019-01-19 16:01:49.923303\t------------------------------\n",
      "2019-01-19 16:01:49.923303\tsetting: {'fastPeriod': 30, 'slowPeriod': 80, 'signalMaPeriod': 15, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 16:01:49.952226\t开始回测\n",
      "2019-01-19 16:01:49.952226\t策略初始化\n",
      "2019-01-19 16:01:50.572568\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 16:01:50.572568\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 16:01:50.572568\t策略初始化完成\n",
      "2019-01-19 16:01:50.572568\t策略启动完成\n",
      "2019-01-19 16:01:50.573567\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 16:01:50.573567\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 16:03:49.529667\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 16:03:49.625413\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 16:03:49.626409\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 16:03:49.906660\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 16:04:22.748881\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 16:04:22.761842\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 16:04:22.775803\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 16:05:00.414201\t回放结束ress: 100%    \n",
      "2019-01-19 16:05:00.415199\t计算按日统计结果\n",
      "2019-01-19 16:05:00.656553\t------------------------------\n",
      "2019-01-19 16:05:00.656553\tsetting: {'fastPeriod': 30, 'slowPeriod': 80, 'signalMaPeriod': 20, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 16:05:00.686472\t开始回测\n",
      "2019-01-19 16:05:00.686472\t策略初始化\n",
      "2019-01-19 16:05:01.266920\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 16:05:01.266920\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 16:05:01.266920\t策略初始化完成\n",
      "2019-01-19 16:05:01.266920\t策略启动完成\n",
      "2019-01-19 16:05:01.266920\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 16:05:01.266920\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 16:05:42.775971\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 16:05:42.847779\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 16:05:42.848776\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 16:05:43.112072\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 16:06:15.706949\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 16:06:15.726897\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 16:06:15.734879\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 16:06:51.903202\t回放结束ress: 100%    \n",
      "2019-01-19 16:06:51.903202\t计算按日统计结果\n",
      "2019-01-19 16:06:52.093691\t------------------------------\n",
      "2019-01-19 16:06:52.093691\tsetting: {'fastPeriod': 40, 'slowPeriod': 40, 'signalMaPeriod': 10, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 16:06:52.128598\t开始回测\n",
      "2019-01-19 16:06:52.128598\t策略初始化\n",
      "2019-01-19 16:06:52.681121\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 16:06:52.681121\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 16:06:52.681121\t策略初始化完成\n",
      "2019-01-19 16:06:52.681121\t策略启动完成\n",
      "2019-01-19 16:06:52.681121\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 16:06:52.681121\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 16:07:33.060192\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 16:07:33.126015\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 16:07:33.127013\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 16:07:33.371361\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 16:08:05.851549\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 16:08:05.865506\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 16:08:05.873485\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 16:08:28.185845\t回放结束ress: 100%    \n",
      "2019-01-19 16:08:28.186845\t计算按日统计结果\n",
      "2019-01-19 16:08:28.186845\t成交记录为空，无法计算逐日回测结果\n",
      "2019-01-19 16:08:28.317494\t------------------------------\n",
      "2019-01-19 16:08:28.318491\tsetting: {'fastPeriod': 40, 'slowPeriod': 40, 'signalMaPeriod': 15, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 16:08:28.348412\t开始回测\n",
      "2019-01-19 16:08:28.348412\t策略初始化\n",
      "2019-01-19 16:08:28.914928\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 16:08:28.915894\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 16:08:28.915894\t策略初始化完成\n",
      "2019-01-19 16:08:28.915894\t策略启动完成\n",
      "2019-01-19 16:08:28.915894\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 16:08:28.915894\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 16:10:36.664435\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 16:10:36.759181\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 16:10:36.760179\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 16:10:37.051401\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 16:11:09.921542\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 16:11:09.943485\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 16:11:09.964427\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 16:11:31.345278\t回放结束ress: 100%    \n",
      "2019-01-19 16:11:31.345278\t计算按日统计结果\n",
      "2019-01-19 16:11:31.345278\t成交记录为空，无法计算逐日回测结果\n",
      "2019-01-19 16:11:31.531780\t------------------------------\n",
      "2019-01-19 16:11:31.531780\tsetting: {'fastPeriod': 40, 'slowPeriod': 40, 'signalMaPeriod': 20, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 16:11:31.562697\t开始回测\n",
      "2019-01-19 16:11:31.562697\t策略初始化\n",
      "2019-01-19 16:11:32.122202\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 16:11:32.122202\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 16:11:32.123199\t策略初始化完成\n",
      "2019-01-19 16:11:32.123199\t策略启动完成\n",
      "2019-01-19 16:11:32.123199\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 16:11:32.123199\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 16:12:13.703060\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 16:12:13.773870\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 16:12:13.773870\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 16:12:14.007247\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 16:12:46.428620\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 16:12:46.441554\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 16:12:46.449532\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 16:13:07.529188\t回放结束ress: 100%    \n",
      "2019-01-19 16:13:07.529188\t计算按日统计结果\n",
      "2019-01-19 16:13:07.529188\t成交记录为空，无法计算逐日回测结果\n",
      "2019-01-19 16:13:07.674798\t------------------------------\n",
      "2019-01-19 16:13:07.675797\tsetting: {'fastPeriod': 40, 'slowPeriod': 60, 'signalMaPeriod': 10, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 16:13:07.707710\t开始回测\n",
      "2019-01-19 16:13:07.707710\t策略初始化\n",
      "2019-01-19 16:13:08.271205\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 16:13:08.271205\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 16:13:08.272202\t策略初始化完成\n",
      "2019-01-19 16:13:08.272202\t策略启动完成\n",
      "2019-01-19 16:13:08.272202\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 16:13:08.272202\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 16:13:48.984382\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 16:13:49.058184\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 16:13:49.058184\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 16:13:49.298542\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 16:14:21.753792\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 16:14:21.771771\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 16:14:21.778725\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 16:14:58.095655\t回放结束ress: 100%    \n",
      "2019-01-19 16:14:58.095655\t计算按日统计结果\n",
      "2019-01-19 16:14:58.447732\t------------------------------\n",
      "2019-01-19 16:14:58.453696\tsetting: {'fastPeriod': 40, 'slowPeriod': 60, 'signalMaPeriod': 15, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 16:14:58.486609\t开始回测\n",
      "2019-01-19 16:14:58.486609\t策略初始化\n",
      "2019-01-19 16:14:59.086008\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 16:14:59.086008\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 16:14:59.086008\t策略初始化完成\n",
      "2019-01-19 16:14:59.086008\t策略启动完成\n",
      "2019-01-19 16:14:59.086008\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 16:14:59.086008\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 16:15:42.838062\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 16:15:42.899896\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 16:15:42.900894\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 16:15:43.153219\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 16:16:15.488791\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 16:16:15.506742\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 16:16:15.516716\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 16:16:53.411427\t回放结束ress: 100%    \n",
      "2019-01-19 16:16:53.411427\t计算按日统计结果\n",
      "2019-01-19 16:16:53.594935\t------------------------------\n",
      "2019-01-19 16:16:53.595932\tsetting: {'fastPeriod': 40, 'slowPeriod': 60, 'signalMaPeriod': 20, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 16:16:53.628845\t开始回测\n",
      "2019-01-19 16:16:53.629843\t策略初始化\n",
      "2019-01-19 16:16:54.290078\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 16:16:54.290078\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 16:16:54.290078\t策略初始化完成\n",
      "2019-01-19 16:16:54.290078\t策略启动完成\n",
      "2019-01-19 16:16:54.291075\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 16:16:54.291075\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 16:19:42.044685\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 16:19:42.149405\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 16:19:42.150403\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 16:19:42.447608\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 16:20:15.238959\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 16:20:15.251925\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 16:20:15.267883\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 16:20:52.244050\t回放结束ress: 100%    \n",
      "2019-01-19 16:20:52.244050\t计算按日统计结果\n",
      "2019-01-19 16:20:52.515355\t------------------------------\n",
      "2019-01-19 16:20:52.516321\tsetting: {'fastPeriod': 40, 'slowPeriod': 80, 'signalMaPeriod': 10, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 16:20:52.547270\t开始回测\n",
      "2019-01-19 16:20:52.547270\t策略初始化\n",
      "2019-01-19 16:20:53.139659\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 16:20:53.140654\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 16:20:53.140654\t策略初始化完成\n",
      "2019-01-19 16:20:53.140654\t策略启动完成\n",
      "2019-01-19 16:20:53.140654\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 16:20:53.140654\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 16:21:34.273707\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 16:21:34.340528\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 16:21:34.341529\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 16:21:34.592855\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 16:22:07.125901\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 16:22:07.149833\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 16:22:07.177758\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 16:22:49.846708\t回放结束ress: 100%    \n",
      "2019-01-19 16:22:49.846708\t计算按日统计结果\n",
      "2019-01-19 16:22:50.052160\t------------------------------\n",
      "2019-01-19 16:22:50.052160\tsetting: {'fastPeriod': 40, 'slowPeriod': 80, 'signalMaPeriod': 15, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 16:22:50.109007\t开始回测\n",
      "2019-01-19 16:22:50.110004\t策略初始化\n",
      "2019-01-19 16:22:50.698432\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 16:22:50.699429\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 16:22:50.699429\t策略初始化完成\n",
      "2019-01-19 16:22:50.699429\t策略启动完成\n",
      "2019-01-19 16:22:50.699429\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 16:22:50.699429\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 16:23:31.699838\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 16:23:31.838468\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 16:23:31.840481\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 16:23:32.166592\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 16:24:05.044712\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 16:24:05.063662\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 16:24:05.074631\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 16:24:47.174102\t回放结束ress: 100%    \n",
      "2019-01-19 16:24:47.175101\t计算按日统计结果\n",
      "2019-01-19 16:24:47.362598\t------------------------------\n",
      "2019-01-19 16:24:47.362598\tsetting: {'fastPeriod': 40, 'slowPeriod': 80, 'signalMaPeriod': 20, 'symbolList': ['EOSUSDT:binance']}\n",
      "仓位字典构造完成 \n",
      "初始仓位: {'EOSUSDT:binance_LONG': 0, 'EOSUSDT:binance_SHORT': 0}\n",
      "2019-01-19 16:24:47.396507\t开始回测\n",
      "2019-01-19 16:24:47.396507\t策略初始化\n",
      "2019-01-19 16:24:48.006876\t数据载入完成, 时间段:[20180623 04:00,20180701 12:00);数据量:11295\n",
      "2019-01-19 16:24:48.006876\t初始化预加载数据成功, 数据长度:11295\n",
      "2019-01-19 16:24:48.006876\t策略初始化完成\n",
      "2019-01-19 16:24:48.006876\t策略启动完成\n",
      "2019-01-19 16:24:48.007875\t回测时间范围:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 16:24:48.007875\t当前回放的时间段:[20180701 12:00,20181201 12:00)\n",
      "2019-01-19 16:25:30.454419\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 16:25:30.523234\t数据载入完成, 时间段:[20180701 12:00,20181201 12:00);数据量:218513\n",
      "2019-01-19 16:25:30.523234\tProgress: 0%    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 16:25:30.772568\t数据载入完成, 时间段:[20180630 05:51,20180703 12:01);数据量:4690\n",
      "2019-01-19 16:27:55.448983\tMongoDB无法连接,当前仅使用本地缓存数据, 请注意数据量。\n",
      "2019-01-19 16:27:55.494771\t数据载入完成, 时间段:[20180524 19:30,20180630 05:50);数据量:46385\n",
      "2019-01-19 16:27:55.510698\tWARNING: 该时间段:[20180703 12:01,20180703 12:30) 数据量为0!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\vnpy_fxdayu-1.1.17-py3.6.egg\\vnpy\\trader\\app\\ctaStrategy\\ctaBacktesting.py\", line 285, in loadHistoryData\n",
      "    if symbol in self.dbClient[self.dbName].collection_names():\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 715, in collection_names\n",
      "    nameOnly=True, **kws)]\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\database.py\", line 674, in list_collections\n",
      "    read_pref) as (sock_info, slave_okay):\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\contextlib.py\", line 81, in __enter__\n",
      "    return next(self.gen)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\mongo_client.py\", line 1099, in _socket_for_reads\n",
      "    server = topology.select_server(read_preference)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 224, in select_server\n",
      "    address))\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 183, in select_servers\n",
      "    selector, server_timeout, address)\n",
      "  File \"C:\\Users\\peng\\Anaconda3\\lib\\site-packages\\pymongo\\topology.py\", line 199, in _select_servers_loop\n",
      "    self._error_message(selector))\n",
      "pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [WinError 10061] 由于目标计算机积极拒绝，无法连接。\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-01-19 16:28:32.206613\t回放结束ress: 100%    \n",
      "2019-01-19 16:28:32.206613\t计算按日统计结果\n",
      "2019-01-19 16:28:32.270442\t------------------------------\n",
      "2019-01-19 16:28:32.270442\t优化结果：\n",
      "2019-01-19 16:28:32.270442\t参数：[\"{'fastPeriod': 40, 'slowPeriod': 60, 'signalMaPeriod': 10, 'symbolList': ['EOSUSDT:binance']}\"]，目标：1.03288683639\n",
      "2019-01-19 16:28:32.270442\t参数：[\"{'fastPeriod': 40, 'slowPeriod': 60, 'signalMaPeriod': 15, 'symbolList': ['EOSUSDT:binance']}\"]，目标：1.03288683639\n",
      "2019-01-19 16:28:32.270442\t参数：[\"{'fastPeriod': 40, 'slowPeriod': 60, 'signalMaPeriod': 20, 'symbolList': ['EOSUSDT:binance']}\"]，目标：1.03288683639\n",
      "2019-01-19 16:28:32.270442\t参数：[\"{'fastPeriod': 30, 'slowPeriod': 80, 'signalMaPeriod': 10, 'symbolList': ['EOSUSDT:binance']}\"]，目标：0.884769784575\n",
      "2019-01-19 16:28:32.270442\t参数：[\"{'fastPeriod': 30, 'slowPeriod': 80, 'signalMaPeriod': 15, 'symbolList': ['EOSUSDT:binance']}\"]，目标：0.884769784575\n",
      "2019-01-19 16:28:32.270442\t参数：[\"{'fastPeriod': 30, 'slowPeriod': 80, 'signalMaPeriod': 20, 'symbolList': ['EOSUSDT:binance']}\"]，目标：0.884769784575\n",
      "2019-01-19 16:28:32.270442\t参数：[\"{'fastPeriod': 30, 'slowPeriod': 60, 'signalMaPeriod': 10, 'symbolList': ['EOSUSDT:binance']}\"]，目标：-0.446513644898\n",
      "2019-01-19 16:28:32.271438\t参数：[\"{'fastPeriod': 30, 'slowPeriod': 60, 'signalMaPeriod': 15, 'symbolList': ['EOSUSDT:binance']}\"]，目标：-0.446513644898\n",
      "2019-01-19 16:28:32.271438\t参数：[\"{'fastPeriod': 30, 'slowPeriod': 60, 'signalMaPeriod': 20, 'symbolList': ['EOSUSDT:binance']}\"]，目标：-0.446513644898\n",
      "2019-01-19 16:28:32.271438\t参数：[\"{'fastPeriod': 40, 'slowPeriod': 80, 'signalMaPeriod': 10, 'symbolList': ['EOSUSDT:binance']}\"]，目标：-1.51035744415\n",
      "2019-01-19 16:28:32.271438\t参数：[\"{'fastPeriod': 40, 'slowPeriod': 80, 'signalMaPeriod': 15, 'symbolList': ['EOSUSDT:binance']}\"]，目标：-1.51035744415\n",
      "2019-01-19 16:28:32.271438\t参数：[\"{'fastPeriod': 40, 'slowPeriod': 80, 'signalMaPeriod': 20, 'symbolList': ['EOSUSDT:binance']}\"]，目标：-1.51035744415\n",
      "2019-01-19 16:28:32.271438\t参数：[\"{'fastPeriod': 30, 'slowPeriod': 40, 'signalMaPeriod': 10, 'symbolList': ['EOSUSDT:binance']}\"]，目标：-1.5489511638\n",
      "2019-01-19 16:28:32.271438\t参数：[\"{'fastPeriod': 30, 'slowPeriod': 40, 'signalMaPeriod': 15, 'symbolList': ['EOSUSDT:binance']}\"]，目标：-1.5489511638\n",
      "2019-01-19 16:28:32.271438\t参数：[\"{'fastPeriod': 30, 'slowPeriod': 40, 'signalMaPeriod': 20, 'symbolList': ['EOSUSDT:binance']}\"]，目标：-1.5489511638\n",
      "2019-01-19 16:28:32.271438\t参数：[\"{'fastPeriod': 20, 'slowPeriod': 60, 'signalMaPeriod': 10, 'symbolList': ['EOSUSDT:binance']}\"]，目标：-2.0628184858\n",
      "2019-01-19 16:28:32.271438\t参数：[\"{'fastPeriod': 20, 'slowPeriod': 60, 'signalMaPeriod': 15, 'symbolList': ['EOSUSDT:binance']}\"]，目标：-2.0628184858\n",
      "2019-01-19 16:28:32.271438\t参数：[\"{'fastPeriod': 20, 'slowPeriod': 60, 'signalMaPeriod': 20, 'symbolList': ['EOSUSDT:binance']}\"]，目标：-2.0628184858\n",
      "2019-01-19 16:28:32.271438\t参数：[\"{'fastPeriod': 20, 'slowPeriod': 80, 'signalMaPeriod': 10, 'symbolList': ['EOSUSDT:binance']}\"]，目标：-2.3671989167\n",
      "2019-01-19 16:28:32.271438\t参数：[\"{'fastPeriod': 20, 'slowPeriod': 80, 'signalMaPeriod': 15, 'symbolList': ['EOSUSDT:binance']}\"]，目标：-2.3671989167\n",
      "2019-01-19 16:28:32.271438\t参数：[\"{'fastPeriod': 20, 'slowPeriod': 80, 'signalMaPeriod': 20, 'symbolList': ['EOSUSDT:binance']}\"]，目标：-2.3671989167\n",
      "2019-01-19 16:28:32.271438\t参数：[\"{'fastPeriod': 20, 'slowPeriod': 40, 'signalMaPeriod': 10, 'symbolList': ['EOSUSDT:binance']}\"]，目标：-2.90176847072\n",
      "2019-01-19 16:28:32.271438\t参数：[\"{'fastPeriod': 20, 'slowPeriod': 40, 'signalMaPeriod': 15, 'symbolList': ['EOSUSDT:binance']}\"]，目标：-2.90176847072\n",
      "2019-01-19 16:28:32.271438\t参数：[\"{'fastPeriod': 20, 'slowPeriod': 40, 'signalMaPeriod': 20, 'symbolList': ['EOSUSDT:binance']}\"]，目标：-2.90176847072\n",
      "耗时：3639.5324087142944\n"
     ]
    }
   ],
   "source": [
    "from vnpy.trader.app.ctaStrategy.ctaBacktesting  import OptimizationSetting\n",
    "# 优化配置\n",
    "from vnpy.trader.app.ctaStrategy import BacktestingEngine\n",
    "import pandas as pd\n",
    "setting = OptimizationSetting()                # 新建一个优化任务设置对象\n",
    "setting.setOptimizeTarget('sharpeRatio')        # 设置优化排序的目标是夏普\n",
    "setting.addParameter('fastPeriod', 20, 40, 10)    # 增加第一个优化参数，起始30，结束50，步进10\n",
    "setting.addParameter('slowPeriod', 40, 80, 20)    # 增加第二个优化参数，起始60，结束30，步进10\n",
    "setting.addParameter('signalMaPeriod', 10, 20, 5)    # 增加第二个优化参数，起始10，结束30，步进5\n",
    "#setting.addParameter('long', 8, 24, 2)\n",
    "setting.addParameter('symbolList', ['EOSUSDT:binance']) \n",
    "\n",
    "import time\n",
    "start = time.time()\n",
    "# 执行单线程优化\n",
    "resultList = engine.runOptimization(adxMacdEosStrategy, setting)\n",
    "# 执行多进程优化一般会改写成py文件然后用多线程运行，提高优化速度。\n",
    "# resultList = engine.runParallelOptimization(MultiFrameMaStrategy, setting)\n",
    "print('耗时：%s' %(time.time()-start))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 486,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "####参数优化"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'from vnpy.trader.app.ctaStrategy.ctaBacktesting  import OptimizationSetting\\n# 优化配置\\nsetting = OptimizationSetting()                # 新建一个优化任务设置对象\\nsetting.setOptimizeTarget(\\'sharpeRatio\\')        # 设置优化排序的目标是夏普\\nsetting.addParameter(\"a\", 1.00, 1.10, 0.01)    # 增加第一个优化参数，起始30，结束50，步进10\\nsetting.addParameter(\"b\", 0.90, 1.00, 0.01)    # 增加第二个优化参数，起始60，结束30，步进10\\n# setting.addParameter(\\'signalMaPeriod\\', 10, 20, 5)    # 增加第二个优化参数，起始10，结束30，步进5\\nsetting.addParameter(\\'symbolList\\', [\\'BTCUSDT:binance\\']) \\nimport time\\nstart = time.time()\\n# 执行单线程优化\\nresultList = engine.runOptimization(MACDRSIStrategy, setting)\\nprint(\\'耗时：%s\\' %(time.time()-start))'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"\"\"from vnpy.trader.app.ctaStrategy.ctaBacktesting  import OptimizationSetting\n",
    "# 优化配置\n",
    "setting = OptimizationSetting()                # 新建一个优化任务设置对象\n",
    "setting.setOptimizeTarget('sharpeRatio')        # 设置优化排序的目标是夏普\n",
    "setting.addParameter(\"a\", 1.00, 1.10, 0.01)    # 增加第一个优化参数，起始30，结束50，步进10\n",
    "setting.addParameter(\"b\", 0.90, 1.00, 0.01)    # 增加第二个优化参数，起始60，结束30，步进10\n",
    "# setting.addParameter('signalMaPeriod', 10, 20, 5)    # 增加第二个优化参数，起始10，结束30，步进5\n",
    "setting.addParameter('symbolList', ['BTCUSDT:binance']) \n",
    "import time\n",
    "start = time.time()\n",
    "# 执行单线程优化\n",
    "resultList = engine.runOptimization(MACDRSIStrategy, setting)\n",
    "print('耗时：%s' %(time.time()-start))\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#import pandas as pd\n",
    "#print(pd.DataFrame(resultList).sort_values(1,  ascending=False))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
